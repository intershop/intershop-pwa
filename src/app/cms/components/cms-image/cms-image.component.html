<ng-container *ngIf="pagelet && pagelet.hasParam('Link'); else image">
  <a [routerLink]="routerLink(pagelet, 'Link')"
    [attr.title]="pagelet.stringParam('LinkTitle')"
    class="teaser-link"
  >
    <ng-container *ngTemplateOutlet="image"></ng-container>
  </a>
</ng-container>

<ng-template #image>
  <img [src]="pagelet.stringParam('Image')"
    [attr.alt]="pagelet.stringParam('AlternateText')"
    [ngClass]="pagelet.stringParam('CSSClass')"
  >
</ng-template>

<!-- TODO: (CMS Component Configuration)
  Picture
-->
