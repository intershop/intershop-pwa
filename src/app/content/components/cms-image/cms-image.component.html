<ng-container *ngIf="getConfigurationParameterValue(pagelet, 'Link'); else image">
  <a [routerLink]="getRouterLink(getConfigurationParameterValue(pagelet, 'Link'))"
    [attr.title]="getConfigurationParameterValue(pagelet, 'LinkTitle')"
    class="teaser-link"
  >
    <ng-container *ngTemplateOutlet="image"></ng-container>
  </a>
</ng-container>

<ng-template #image>
  <img [src]="getImagePath(getConfigurationParameterValue(pagelet, 'Image'))"
    [attr.alt]="getConfigurationParameterValue(pagelet, 'AlternateText')"
    [ngClass]="getConfigurationParameterValue(pagelet, 'CSSClass')"
  >
</ng-template>

<!-- TODO: (CMS Component Configuration)
  Picture
-->
