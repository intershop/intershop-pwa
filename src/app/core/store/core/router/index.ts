// API to access ngrx core router state
export * from './router.selectors';
export * from './router.operators';
