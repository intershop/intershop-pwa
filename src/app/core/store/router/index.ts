export * from './router.actions';
export * from './router.selectors';
export { CoreState } from '../core.state';
