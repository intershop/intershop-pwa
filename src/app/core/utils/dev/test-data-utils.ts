import { CategoryTree, CategoryTreeHelper } from 'ish-core/models/category-tree/category-tree.model';
import { Category } from 'ish-core/models/category/category.model';
import { ContentPageTreeHelper } from 'ish-core/models/content-page-tree/content-page-tree.helper';
import { ContentPageTree, ContentPageTreeElement } from 'ish-core/models/content-page-tree/content-page-tree.model';

export function categoryTree(categories?: Category[]): CategoryTree {
  const tree = CategoryTreeHelper.empty();
  if (categories) {
    return categories.reduce((acc, cat) => CategoryTreeHelper.add(acc, cat), tree);
  }
  return tree;
}

export function pageTree(elements?: ContentPageTreeElement[]): ContentPageTree {
  const tree = ContentPageTreeHelper.empty();
  if (elements) {
    return elements.reduce((acc, cat) => ContentPageTreeHelper.add(acc, cat), tree);
  }
  return tree;
}
