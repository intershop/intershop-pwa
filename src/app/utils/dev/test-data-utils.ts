import { CategoryTree, CategoryTreeHelper } from 'ish-core/models/category-tree/category-tree.model';
import { Category } from 'ish-core/models/category/category.model';

export function categoryTree(categories?: Category[]): CategoryTree {
  const tree = CategoryTreeHelper.empty();
  if (categories) {
    return categories.reduce((acc, cat) => CategoryTreeHelper.add(acc, cat), tree);
  }
  return tree;
}
