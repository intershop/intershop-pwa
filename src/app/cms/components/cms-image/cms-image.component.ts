import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ContentViewHelper } from 'ish-core/models/content-view/content-view.helper';
import { ContentPageletView } from 'ish-core/models/content-view/content-views';
import { CMSComponent } from '../../models/cms-component/cms-component.model';

@Component({
  selector: 'ish-cms-image',
  templateUrl: './cms-image.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CMSImageComponent implements CMSComponent {
  @Input() pagelet: ContentPageletView;

  routerLink = ContentViewHelper.getRouterLink;
}
