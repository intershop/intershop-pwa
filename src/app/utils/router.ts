import { ActivatedRoute, ChildActivationEnd, Router } from '@angular/router';
import { merge, Observable } from 'rxjs';
import { distinctUntilChanged, filter, pluck } from 'rxjs/operators';

/**
 * resolves data from child routes
 */
export function resolveChildRouteData<T>(route: ActivatedRoute, router: Router, key: string): Observable<T> {
  return merge(
    // get first data from injected activated route
    route.firstChild.data.pipe(pluck(key)),
    // listen to router events to get other child updates
    router.events.pipe(
      filter(event => event instanceof ChildActivationEnd),
      pluck('snapshot', 'data', key),
      filter((x: T) => !!x)
    )
  ).pipe(distinctUntilChanged());
}
