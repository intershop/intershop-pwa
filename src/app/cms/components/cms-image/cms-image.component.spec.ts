import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { createSimplePageletView } from 'ish-core/utils/dev/test-data-utils';

import { CMSImageComponent } from './cms-image.component';

describe('Cms Image Component', () => {
  let component: CMSImageComponent;
  let fixture: ComponentFixture<CMSImageComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [CMSImageComponent],
      imports: [RouterTestingModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CMSImageComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    const pagelet = {
      definitionQualifiedName: 'fq',
      id: 'id',
      configurationParameters: {
        Image: 'http://example.com/foo/bar.png',
        AlternateText: 'foo',
        CSSClass: 'foo',
        Link: 'link://foo',
        LinkTitle: 'bar',
      },
    };
    component.pagelet = createSimplePageletView(pagelet);
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });
});
