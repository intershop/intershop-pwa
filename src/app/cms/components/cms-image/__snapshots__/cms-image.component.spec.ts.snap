// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cms Image Component should be created 1`] = `
<div
  id="root0"
  ng-version="7.1.2"
>
  
  
  <a
    class="teaser-link"
    href="/home"
    ng-reflect-router-link="/home"
    title="bar"
  >
    
    <img
      alt="foo"
      class="foo"
      ng-reflect-ng-class="foo"
      src="http://www.example.org/foo/-bar.png"
    />
  </a>
  
</div>
`;
